test/e2e/retry.go

```diff
@@ -36,7 +37,7 @@ const (
 // retryWithTimeout retries a function that returns an error until a timeout is reached
 func retryWithTimeout(interval, timeout time.Duration, fn func() error) error {
 	var pollError error
-	err := wait.PollImmediate(interval, timeout, func() (bool, error) {
+	err := wait.PollUntilContextTimeout(context.TODO(), interval, timeout, true, func(context.Context) (bool, error) {
```

Comment:
is there a reason you used `PollUntilContextTimeout` instead of `PollWithContextTimeout` here? I see 

> // Deprecated: This method does not return errors from context, use PollWithContextTimeout.
// Note that the new method will no longer return ErrWaitTimeout and instead return errors
// defined by the context package. Will be removed in a future release.

in the wait package

Reply:
AFAICT that's actually a typo, only `PollUntilContextTimeout` seems to exist: https://pkg.go.dev/k8s.io/apimachinery@v0.27.2/pkg/util/wait#PollUntilContextCancel

Reply:
talked offline `PollWithContextTimeout` doesn't exist, that code comment is likely a typo